boards/imix/src/power.rs

```diff
@@ -0,0 +1,106 @@
+extern crate kernel;
+extern crate sam4l;
+
+use kernel::hil::Controller;
+use sam4l::gpio::{PA, PB, PC};
+use sam4l::gpio::GPIOPin;
+use sam4l::gpio::PeripheralFunction;
+use sam4l::gpio::PeripheralFunction::{A, B};
+
+type DetachablePin = (&'static GPIOPin, Option<PeripheralFunction>);
+
+trait Detachable {
```

Comment:
This trait is not actually being used anywhere, since the only place you use it's methods, you actually know the exact type (`DetachablePin`)